### ISSUE TITLE
warn about :click being a possible mistake ( user meant to use @click)

### ISSUE THREAD

**Issue by sktguha** [vuejs/eslint-plugin-vue#740]
**Please describe what the rule should do:**

If the programmer is using :click (instead of @click) to try to capture click event the rule should warn the user that :click will create a data attribute instead of listening in on the event handler , which they probably intended

**What category should the rule belong to?**

- [ ] Enforces code style
- [X] Warns about a potential error
- [ ] Suggests an alternate way of doing something
- [ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**
```vue
     <button :click="doSearchAction()">Search</button>
```
in the example above, it is correct code only, but this will result in an attribute called click created on the button with the value the result of doSearchAction

**Additional context**

Infact this rule can be extended for any dom events like focus,hover etc

I can start work on this if this is green lighted

**Comment by sktguha** [comment]
Ya that's what I meant . I mean I used :click on an element and called a function. And I was wondering for sometime why that is not working properly. So I want that others won't make the same mistake.

**Comment by michalsnik** [comment]
We might allow users to whitelist props, that they are certain about @sktguha

**Comment by sktguha** [comment]
Ya that could work too.
Hmm ok then let me know what final decision is taken on this.

**Comment by sktguha** [comment]
let me know in case of any update

**Comment by sktguha** [comment]
hello anything @michalsnik ?

**Comment by sktguha** [comment]
hi, anyone interested in this ? I have often run into this issue so the eslint rule will be a good addition
